### ISSUE TITLE
Does ForgeUI support Flux Fill models (Flux-inpainting)?

### ISSUE THREAD

**Issue by Renaldas111** [lllyasviel/stable-diffusion-webui-forge#2486]
It looks all these models throw the same error:

RuntimeError: mat1 and mat2 shapes cannot be multiplied (4032x64 and 384x3072)
mat1 and mat2 shapes cannot be multiplied (4032x64 and 384x3072)

**Comment by danividalg** [comment]
+1

**Comment by morozig** [comment]
Same here: `RuntimeError: mat1 and mat2 shapes cannot be multiplied (3072x64 and 384x3072)`

**Comment by Renaldas111** [comment]
All previous inpainting models (SD1.5, SDXL) had a different latent shape, they had 9 channels instead of "typical" 4 channels. Probably this mismatch in tensors is due to the same different shape of inpainting latent tensor?

**Comment by morozig** [comment]
I feel like this repo uses model configs from black-forest-labs to determine model params and dimensions. But flux1-dev and flux1-fill have different shapes in unet that's why matmul throws an error about  `64 and 384`. Probably we would need to add a full new config of flux1-fill and new model type or something to make this work. Hopefully someone knows how to do it...
![matmul-error1](https://github.com/user-attachments/assets/bedd7d5d-0e58-43de-b1ec-ad1cd7bd630b)
![matmul-error2](https://github.com/user-attachments/assets/a67f4262-138d-49cb-9041-a74f7a0c689c)
![matmul-error3](https://github.com/user-attachments/assets/22844795-2763-44d0-88f6-50219181dc17)

**Comment by danilomaiaweb** [comment]
Would it work this way below?

{
  "_class_name": "FluxTransformer2DModel",
  "_diffusers_version": "0.32.0.dev0",
  "attention_head_dim": 128,
  "axes_dims_rope": [
    16,
    56,
    56
  ],
  "guidance_embed": true,
  "in_channels": [
    64,
    384
  ],
  "joint_attention_dim": 4096,
  "num_attention_heads": 24,
  "num_layers": 19,
  "num_single_layers": 38,
  "out_channels": 64,
  "patch_size": 1,
  "out_projection_dim": 768
}

**Comment by morozig** [comment]
> Would it work this way below?
> 
> { "_class_name": "FluxTransformer2DModel", "_diffusers_version": "0.32.0.dev0", "attention_head_dim": 128, "axes_dims_rope": [ 16, 56, 56 ], "guidance_embed": true, "in_channels": [ 64, 384 ], "joint_attention_dim": 4096, "num_attention_heads": 24, "num_layers": 19, "num_single_layers": 38, "out_channels": 64, "patch_size": 1, "out_projection_dim": 768 }

I've tried quickly replacing it inside `backend/huggingface/black-forest-labs/FLUX.1-dev/transformer/config.json` but it did nothing. This problem would surely require a lot more work from someone with knowledge of this repo.

**Comment by parrydigm** [comment]
> I feel like this repo uses model configs from black-forest-labs to determine model params and dimensions. But flux1-dev and flux1-fill have different shapes in unet that's why matmul throws an error about `64 and 384`. Probably we would need to add a full new config of flux1-fill and new model type or something to make this work.

Thank heavens I found this post, I was tearing my hair out trying different VAE to get it to work. I'm fortunate that I can use DrawThings (MacOS) for flux fill support until this is resolved, but the ol' M2 is not quite at the same speed level as a 4070 Super 🤣

**Comment by ride5k** [comment]
is this still an issue?  interested in trying out flux inpainting...

**Comment by parrydigm** [comment]
> is this still an issue? interested in trying out flux inpainting...

Yep. I believe Forge is more or less on life support now and we may not see new feature additions. The Stable Diffusion subreddit seems to suggest Swarm UI as worthy of trying out but I’ve had a cognitive error with its UI 🤣
